File: Day23_Koko_Eating_Bananas/Koko_Eating_Bananas.py
import math
class Solution(object):
    def minEatingSpeed(self, piles, h):
        """
        :type piles: List[int]
        :type h: int
        :rtype: int
        """
        mini=1
        maxi=max(piles)
        ans=maxi
        while mini<=maxi:
            mid=(mini+maxi)//2
            count=0
            for i in piles:
                count+=math.ceil(float(i)/mid)
            if count<=h:
                maxi=mid-1
                ans=mid
            else:
                mini=mid+1
        return ans
        
        
        

File: Day23_Koko_Eating_Bananas/Koko_Eating_Bananas.java

class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int mini=1;
        int maxi=0;
        for(int i=0;i<piles.length;i++){
            if(piles[i]>maxi){
                maxi=piles[i];
            }
        }
        int ans=maxi;
        while(mini<=maxi){

            int mid=(mini+(maxi-mini)/2);
            double count=0;
            for(int i=0;i<piles.length;i++){
                count+=((piles[i]+mid-1)/mid);
                
            }
            if(count<=h){
                maxi=mid-1;
                ans=mid;
                System.out.println(mid+" "+count+" "+h);
            }
            else{
                mini=mid+1;
            }

        }
        return ans;

    }
}
